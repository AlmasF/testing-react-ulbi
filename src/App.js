import { useState, useEffect } from 'react';
import './App.css';

function App() {
  const [data, setData] = useState(null);
  const [toggle, setToggle] = useState(false);
  const [value, setValue] = useState('');

  const onClick = () => setToggle(prev => !prev);

  useEffect(() => {
    setTimeout(() => {
      setData({});
    }, 100);
  }, []);

  return (
    <div className="App">
      { 
        toggle === true 
        && <div data-testId='toggle-elem'>toggle</div>
      }
      { 
        data && 
        <div style={{color: 'red'}}>data</div> 
      }
      <h1 data-testId="value-elem">{ value }</h1>
      <button
        data-testId="toggle-btn"
        onClick={onClick}
      >
        click me
      </button>
      <input
        type="text"
        placeholder="input value"
        onChange={e => setValue(e.target.value)}
      />
    </div>
  );
}

export default App;
