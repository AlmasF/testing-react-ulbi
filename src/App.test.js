import { render } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';

describe('TEST APP', () => {
  test('renders elements', () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
  });

//   test('renders learn react link', async () => {
//     render(<App />);
//     // const helloWorldElem = screen.queryByText(/hello2 world/i);
//     // expect(helloWorldElem).toBeNull();
//     screen.debug();
//     const helloWorldElem = await screen.findByText(/data/i);
//     expect(helloWorldElem).toBeInTheDocument();
//     expect(helloWorldElem).toHaveStyle({color: 'red'});
//     screen.debug();
//   });

//   test('CLICK EVENT', async () => {
//     render(<App />);
//     screen.debug();
//     const btn = screen.getByTestId('toggle-btn');
//     expect(screen.queryByTestId('toggle-elem')).toBeNull();
//     fireEvent.click(btn);
//     expect(screen.queryByTestId('toggle-elem')).toBeInTheDocument();
//     fireEvent.click(btn);
//     expect(screen.queryByTestId('toggle-elem')).toBeNull();
//   });

//   test('INPUT EVENT', async () => {
//     render(<App />);
//     screen.debug();
//     const input = screen.getByPlaceholderText(/input value/i);
//     expect(screen.queryByTestId('value-elem')).toContainHTML('');
//     // Искусственное событие
//     // fireEvent.input(input, {
//     //   target: { value: '123123' },
//     // });
//     // Близко к пользователям, обрабатываются события нажатия клавиш и т д
//     userEvent.type(input, '123123');
//     expect(screen.queryByTestId('value-elem')).toContainHTML('123123');
//   });
});
