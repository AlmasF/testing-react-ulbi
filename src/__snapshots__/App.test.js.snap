// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP renders elements 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;
